lib/vssh/libssh.c

```diff
@@ -336,7 +336,13 @@ static int myssh_is_known(struct connectdata *conn)
   struct curl_khkey knownkey;
 #endif
 
-#if LIBSSH_VERSION_INT >= SSH_VERSION_INT(0,8,0)
+  /* libssh 0.8.0 renamed ssh_get_publickey and marked it as deprecated.
+   * Ubuntu 18.04 however ships with a pre-release version (commit 94fa1e38)
+   * that also includes that change. As workaround, check for a symbol from
+   * libssh/sftp.h that was added after 0.7.0 to avoid deprecation warnings.
+   */
+#if (LIBSSH_VERSION_INT >= SSH_VERSION_INT(0, 8, 0)) || \
+    (LIBSSH_VERSION_INT == SSH_VERSION_INT(0, 7, 0) && defined(SSH_S_IFMT))
```

Comment:
Can you make this into a separate PR instead? Makes it easier to get the right set of people to review the separate changes.

Reply:
Will do, but this PR will still depend on that change to avoid breaking CI after the upgrade.

Reply:
Right, so let's try to get that landed first...

Reply:
These two commits are now in https://github.com/curl/curl/pull/5372 as well, but they are still included here to allow CI to pass.